### PR TITLE
#7496 Don`t allow upload of files with extension .part

### PR DESCRIPTION
**Pull request by IljaN**
## Description
Don`t allow files with extension .part
 
## Related Issue
 #7496

## How Has This Been Tested?

Manually test uploading .part files

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/private/legacy/util.php [hunk 1]
```diff
@@ -1376,6 +1376,12 @@ public static function isValidFileName($file) {
 		if (\OC\Files\Filesystem::isIgnoredDir($trimmed)) {
 			return false;
 		}
+
+		$path_parts = pathinfo($trimmed);
+		if ($path_parts['extension'] === 'part') {
```
Comment (PVince81) [inline lib/private/legacy/util.php]: Please use the constant that @felixheidecke has introduced here for you: https://github.com/owncloud/core/pull/27149/files#diff-6cc3350c00dd13da023086587cce8bfdR64

**Comment by mmattel** [comment]
@IljaN any update on this?

**Comment by PVince81** [comment]
@mmattel feel free to take over if you have time, else I might finish this tomorrow.
@IljaN is busy with other stuff currently.

**Comment by PVince81** [comment]
Adjusted to use the regexp and updated unit tests.

Curious to see how many other things will break...

**Comment by mmattel** [comment]
Q: in config.sample.php we have blacklisted files. Hardcoded exact names with a default.
Why not having a parameter catching those in the same way, this would also ease up documentaion.

**Comment by PVince81** [comment]
I think we can't reuse these settings for two reasons:

- files APIs like View, etc still need to be able to write part files on disk and read them instead of skipping them
- if we make it a setting, then it means some people would try to disable them and trigger the issues we're trying to avoid by forbidding them

**Comment by mmattel** [comment]
Misunderstanding...

Did not say reuse the parameter, but reuse the idea with a NEW parameter. Blacklisted has hardcoded .htaccess and that can´t be changed by an admin. You can only add. It is a full filename.

My idea is based on Blacklisted, a new parameter like Blacklisted_Filepart which has hardcoded behind .part or whatever we need, but also enables future types. By having hat, we automatically have a documentation, tranparency and the inability to change the defaults because of hardcoding.

**Comment by PVince81** [comment]
> inability to change the defaults because of hardcoding

In the case of part files we do not want people to mess with these, so hard-coding is fine for this I think.

If we want customization for additional illegal file patterns, then I'd consider this a new feature and then yes we could add a new config option for that.

**Comment by PVince81** [comment]
@jvillafanez @mmattel review ?

**Comment by davitol** [comment]
Error warning ""FileName.part" is not an allow filetype is kept showing and the user has to close the yellow warning. Not sure if we want that behaviour. If yes, 👍

**Pull request merged**